pages/new-submission.tsx

```diff
@@ -1,9 +1,85 @@
+import { z } from 'Zod'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { useMutation } from '@tanstack/react-query'
+import { SubmitHandler, useForm } from 'react-hook-form'
+import toast, { Toaster } from 'react-hot-toast'
 import { MainLayout } from '../components/layouts/mainLayout'
+import { BackButton } from '../components/shared/BackButton'
+import { BaseButton } from '../components/shared/BaseButton'
+import { BaseInput } from '../components/shared/BaseInput'
+import { BaseTextArea } from '../components/shared/BaseTextArea'
+import { storeSubmission } from '../api/patient'
+
+const submissionSchema = z.object({
+  title: z.string().min(1, { message: 'Title is required' }),
+  symptoms: z.string().min(10, { message: 'Symptoms must be at least 10 characters' }),
+})
+
+export type FormValues = z.infer<typeof submissionSchema>
 
 export default function NewSubmissionPage() {
+  const {
+    reset,
+    formState: { errors },
+    register,
+    handleSubmit,
+  } = useForm<FormValues>({ resolver: zodResolver(submissionSchema) })
+
+  const { mutate } = useMutation({
+    mutationFn: storeSubmission,
+    onSuccess: () => {
+      toast.success('Submission successfully send', {
+        position: 'top-right',
+      })
+      reset()
+    },
+    onError: (error: any) => {
+      toast.error(error.response.data.message, {
+        position: 'top-right',
+      })
+    },
+  })
+
+  const onSubmit: SubmitHandler<FormValues> = data => {
```

Comment:
I believe you could just declare this inline and save yourself from having to declare the types by letting Typescript infer the types